coderd/notifications/notifications_test.go

```diff
@@ -1285,6 +1330,49 @@ func (f *fakeHandler) Dispatcher(payload types.MessagePayload, _, _ string) (dis
 	}, nil
 }
 
+type goldenFileHandler struct {
+	t                 *testing.T
+	updateGoldenFiles bool
+}
+
+func (f *goldenFileHandler) Dispatcher(payload types.MessagePayload, title, body string) (dispatch.DeliveryFunc, error) {
+	return func(_ context.Context, _ uuid.UUID) (retryable bool, err error) {
+		partialName := strings.Split(f.t.Name(), "/")[1]
+		goldenFile := filepath.Join("testdata", "rendered-templates", partialName+".json.golden")
+
+		// UserIDs change on every test run. We need to set it to a known value to compare golden files.
+		payload.UserID = "00000000-0000-0000-0000-000000000000"
+
+		// The following is done by the stmp handler during dispatch.
+		// By doing it here we get an accurate golden representation of
+		// what would have been sent. We don't need to maintain parity
+		// with how the SMTP handler does it. We just need an accurate
+		// representation of the payload, and the rendered body and title.
+		payload.Labels["_body"] = body
+		payload.Labels["_title"] = title
+
+		payloadJSON, err := json.MarshalIndent(payload, "", "  ")
+		require.NoError(f.t, err, "want no error marshaling payload to JSON")
+
+		if *updateGoldenFiles {
```

Comment:
Note to self: This should either refer to the struct attribute or we should remove the struct attribute